plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt

```diff
@@ -208,26 +213,48 @@ class CodeTransformChatController(
         val moduleVirtualFile: VirtualFile = modulePath.toVirtualFile() as VirtualFile
         val moduleName = context.project.getModuleOrProjectNameForFile(moduleVirtualFile)
 
-        codeTransformChatHelper.run {
-            addNewMessage(buildUserSelectionSummaryChatContent(moduleName))
-            addNewMessage(buildCompileLocalInProgressChatContent())
-        }
+        codeTransformChatHelper.addNewMessage(buildUserSelectionSummaryChatContent(moduleName))
 
-        // this should never throw the RuntimeException since invalid JDK case is already handled in previous validation step
-        val moduleJdkVersion = ModuleUtil.findModuleForFile(moduleVirtualFile, context.project)?.tryGetJdk(context.project)
-        logger.info { "Found project JDK version: ${context.project.tryGetJdk()}, module JDK version: $moduleJdkVersion. Module JDK version prioritized." }
-        val sourceJdk = moduleJdkVersion ?: context.project.tryGetJdk() ?: throw RuntimeException("Unable to determine source JDK version")
+        val sourceJdk = getSourceJdk(moduleVirtualFile)
 
         val selection = CustomerSelection(
             moduleVirtualFile,
             sourceJdk,
-            JavaSdkVersion.JDK_17
+            JavaSdkVersion.JDK_17,
         )
 
+        // Create and set a session
+        val session = codeModernizerManager.createCodeModernizerSession(selection, context.project)
+        codeModernizerManager.codeTransformationSession = session
```

Comment:
i would have expected `createCodeModernizerSession` to automatically set the current session instead of relying on the caller to mutate this